### PR TITLE
Allow to use cron with autorestart=false scripts

### PR DESCRIPTION
**Pull request by soyuka**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #2487  
| License       | MIT
| Doc PR        | na

To test:
```
./bin/pm2 start -c '* * * * *' --no-autorestart --no-daemon test/fixtures/cron_no_autorestart.js
```

Wait at least 1 min

### CODE REVIEW

**Comment by Unitech** [comment]
Wise fix
It allows to use cron on stopped apps + on any kind of process besides node.js

**Comment by soyuka** [comment]
> Wise fix
> It allows to use cron on stopped apps + on any kind of process besides node.js

And fixed a 3 years old issue :fireworks:

File: lib/Worker.js [hunk 1]
```diff
@@ -20,6 +22,34 @@ module.exports = function(God) {
...
+    if (!proc ||
+        !proc.pm2_env ||
+        proc.pm2_env.pm_id === undefined ||
+        !proc.pm2_env.cron_restart ||
+        God.CronJobs[proc.pm2_env.pm_id] !== undefined)
+      return;
+
+    console.log('[PM2][WORKER] Registering a cron job on:', proc.pm2_env.pm_id);
+
+    God.CronJobs[proc.pm2_env.pm_id] = new cronJob({
```
Comment (vmarchaud) [inline lib/Worker.js]: I believe a Map should be better here

Comment (soyuka) [reply lib/Worker.js]: Are maps supported by node 4?

Comment (vmarchaud) [reply lib/Worker.js]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map#Browser_compatibility
Since node 0.12

**Comment by soyuka** [comment]
> Programmatic tests are missing, checkout https://github.com/Unitech/pm2/blob/master/test/programmatic/exp_backoff_restart_delay.mocha.js

I can but the test will have to wait at least 1 whole minute to check that the cron did his job.

**Pull request merged**